exporters/trace/elasticsearch/src/main/java/io/opencensus/exporter/trace/elasticsearch/ElasticsearchTraceExporter.java

```diff
@@ -0,0 +1,244 @@
+          elasticsearchTraceConfiguration.getElasticsearchIndex() != null,
+          "Elasticsearch index not specified");
+      Preconditions.checkArgument(
+          elasticsearchTraceConfiguration.getElasticsearchType() != null,
+          "Elasticsearch type not specified");
+      Preconditions.checkArgument(
+          elasticsearchTraceConfiguration.getElasticsearchUrl() != null,
+          "Elasticsearch URL not specified");
+      handler = new ElasticsearchTraceHandler(elasticsearchTraceConfiguration);
+      register(Tracing.getExportComponent().getSpanExporter(), handler);
+    }
+  }
+
+  /**
+   * Registers the ElasticsearchTraceExporter.
+   *
+   * @param spanExporter instance of the {@link SpanExporter} registered.
+   * @param handler instance of the {@link Handler} registered.
+   */
+  public static void register(SpanExporter spanExporter, Handler handler) {
+    spanExporter.registerHandler(REGISTERED_TRACE_EXPORTER_NAME, handler);
+  }
+
+  /**
+   * Unregisters the ElasticsearchTraceExporter from OpenCensus.
+   *
+   * @throws IllegalStateException if ElasticsearchTraceExporter is already unregistered.
+   * @since 0.20.0
+   */
+  public static void unregister() {
+    synchronized (monitor) {
+      Preconditions.checkState(
+          handler != null,
+          "Can't unregister Elasticsearch Trace Exporter which is not registered.");
+      unregister(Tracing.getExportComponent().getSpanExporter());
+      handler = null;
+    }
+  }
+
+  /**
+   * Unregisters the ElasticsearchTraceExporter from OpenCensus.
+   *
+   * @param spanExporter the instance of the {@code SpanExporter} to be unregistered.
+   * @since 0.20.0
+   */
+  public static void unregister(SpanExporter spanExporter) {
+    spanExporter.unregisterHandler(REGISTERED_TRACE_EXPORTER_NAME);
+  }
+
+  static final class ElasticsearchTraceHandler extends SpanExporter.Handler {
+
+    private final ElasticsearchTraceConfiguration elasticsearchTraceConfiguration;
+    private final String appName;
+    private final URL indexUrl;
+
+    public ElasticsearchTraceHandler(
+        ElasticsearchTraceConfiguration elasticsearchTraceConfiguration)
+        throws MalformedURLException {
+
+      this.elasticsearchTraceConfiguration = elasticsearchTraceConfiguration;
+      StringBuilder sb = new StringBuilder();
+      sb.append(elasticsearchTraceConfiguration.getElasticsearchUrl()).append("/");
+      sb.append(elasticsearchTraceConfiguration.getElasticsearchIndex()).append("/");
+      sb.append(elasticsearchTraceConfiguration.getElasticsearchType()).append("/");
+      indexUrl = new URL(sb.toString());
+      appName = elasticsearchTraceConfiguration.getAppName();
+    }
+
+    /**
+     * Handles exporting of traces in {@code ElasticsearchTraceExporter}.
+     *
+     * @param spanDataList Collection of {@code SpanData} to be exported.
+     */
+    @Override
+    public void export(Collection<SpanData> spanDataList) {
+      Scope scope =
```

Comment:
nit: consider using try-with-resource to simplify the code, since this exporter requires Java 7+.